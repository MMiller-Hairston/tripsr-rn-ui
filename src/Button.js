import React from 'react';
import PropTypes from 'prop-types';
import { Text, TouchableOpacity } from 'react-native';

const applyStyle = style => {
  switch (style) {
    case 'primary':
      return styles.buttonPrimary;
    case 'facebook':
      return styles.facebook;
    case 'google':
      return styles.google;
    default:
      return styles.button;
  }
};

const Button = ({ text, onPress, type }) => {
  return (
    <TouchableOpacity onPress={onPress} style={applyStyle(type)}>
      <Text style={styles.buttonText}>{text}</Text>
    </TouchableOpacity>
  );
};

Button.propTypes = {
  text: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

const styles = {
  colorWhite: {
    color: '#fff',
  },
  buttonPrimary: {
    marginLeft: 20,
    marginRight: 20,
    marginTop: 10,
    marginBottom: 10,
    paddingTop: 20,
    paddingBottom: 20,
    paddingRight: 10,
    paddingLeft: 10,
    alignItems: 'center',
    backgroundColor: '#62C6BB',
    borderRadius: 3,
  },
  button: {
    marginLeft: 20,
    marginRight: 20,
    marginTop: 10,
    marginBottom: 10,
    paddingTop: 20,
    paddingBottom: 20,
    paddingRight: 10,
    paddingLeft: 10,
    alignItems: 'center',
    borderColor: '#fff',
    backgroundColor: 'transparent',
    borderWidth: 1,
    borderRadius: 3,
  },
  facebook: {
    marginLeft: 20,
    marginRight: 20,
    marginTop: 10,
    marginBottom: 10,
    paddingTop: 20,
    paddingBottom: 20,
    paddingRight: 10,
    paddingLeft: 10,
    alignItems: 'center',
    backgroundColor: '#3b5998',
    borderRadius: 3,
  },
  google: {
    marginLeft: 20,
    marginRight: 20,
    marginTop: 10,
    marginBottom: 10,
    paddingTop: 20,
    paddingBottom: 20,
    paddingRight: 10,
    paddingLeft: 10,
    alignItems: 'center',
    backgroundColor: '#df4a32',
    borderRadius: 3,
  },
  buttonText: {
    color: 'white',
    fontWeight: '600',
    fontSize: 18,
    textShadowColor: 'black',
    textShadowRadius: 0.4,
    textShadowOffset: { width: 0, height: 0.25 },
  },
};

export default Button;
