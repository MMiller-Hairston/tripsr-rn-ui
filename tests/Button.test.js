import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../src/Button';

test('Button renders correctly', () => {
  const tree = renderer
    .create(<Button text="Test" onPress={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button renders correctly with primary type', () => {
  const tree = renderer
    .create(<Button text="Test" onPress={() => {}} type="primary" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button renders correctly with facebook type', () => {
  const tree = renderer
    .create(<Button text="Test" onPress={() => {}} type="facebook" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button renders correctly with google type', () => {
  const tree = renderer
    .create(<Button text="Test" onPress={() => {}} type="google" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
