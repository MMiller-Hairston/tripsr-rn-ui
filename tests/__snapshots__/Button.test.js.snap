// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderColor": "#fff",
      "borderRadius": 3,
      "borderWidth": 1,
      "marginBottom": 10,
      "marginLeft": 20,
      "marginRight": 20,
      "marginTop": 10,
      "opacity": 1,
      "paddingBottom": 20,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 20,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 18,
        "fontWeight": "600",
        "textShadowColor": "black",
        "textShadowOffset": Object {
          "height": 0.25,
          "width": 0,
        },
        "textShadowRadius": 0.4,
      }
    }
  >
    Test
  </Text>
</View>
`;

exports[`Button renders correctly with facebook type 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3b5998",
      "borderRadius": 3,
      "marginBottom": 10,
      "marginLeft": 20,
      "marginRight": 20,
      "marginTop": 10,
      "opacity": 1,
      "paddingBottom": 20,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 20,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 18,
        "fontWeight": "600",
        "textShadowColor": "black",
        "textShadowOffset": Object {
          "height": 0.25,
          "width": 0,
        },
        "textShadowRadius": 0.4,
      }
    }
  >
    Test
  </Text>
</View>
`;

exports[`Button renders correctly with google type 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#df4a32",
      "borderRadius": 3,
      "marginBottom": 10,
      "marginLeft": 20,
      "marginRight": 20,
      "marginTop": 10,
      "opacity": 1,
      "paddingBottom": 20,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 20,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 18,
        "fontWeight": "600",
        "textShadowColor": "black",
        "textShadowOffset": Object {
          "height": 0.25,
          "width": 0,
        },
        "textShadowRadius": 0.4,
      }
    }
  >
    Test
  </Text>
</View>
`;

exports[`Button renders correctly with primary type 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#62C6BB",
      "borderRadius": 3,
      "marginBottom": 10,
      "marginLeft": 20,
      "marginRight": 20,
      "marginTop": 10,
      "opacity": 1,
      "paddingBottom": 20,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 20,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 18,
        "fontWeight": "600",
        "textShadowColor": "black",
        "textShadowOffset": Object {
          "height": 0.25,
          "width": 0,
        },
        "textShadowRadius": 0.4,
      }
    }
  >
    Test
  </Text>
</View>
`;

exports[`Button renders correctly with primary type 2`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3b5998",
      "borderRadius": 3,
      "marginBottom": 10,
      "marginLeft": 20,
      "marginRight": 20,
      "marginTop": 10,
      "opacity": 1,
      "paddingBottom": 20,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 20,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 18,
        "fontWeight": "600",
        "textShadowColor": "black",
        "textShadowOffset": Object {
          "height": 0.25,
          "width": 0,
        },
        "textShadowRadius": 0.4,
      }
    }
  >
    Test
  </Text>
</View>
`;
